File: lib/core/data/mapper/recipe/ingredient_mapper.dart
import 'package:flutter_recipe_app/core/data/dto/recipe/ingredient_dto.dart';
import 'package:flutter_recipe_app/core/data/dto/recipe/ingredients_dto.dart';
import 'package:flutter_recipe_app/core/domain/model/recipe/ingredient.dart';

extension IngredientsDtoToModel on IngredientsDto {
  Ingredient toModel() {
    return Ingredient(
      id: _initId(ingredient?.id),
      name: ingredient?.name ?? '',
      imageUrl: ingredient?.image ?? '',
      weight: amount?.toDouble() ?? 0,
    );
  }

  String _initId(int? id) {
    return id?.toString() ?? Ingredient.invalidId;
  }
}

extension IngredientsModelToDto on Ingredient {
  IngredientsDto toDto() {
    return IngredientsDto(
      ingredient: IngredientDto(
        id: int.tryParse(id) ?? -1,
        name: name,
        image: imageUrl,
      ),
      amount: weight.toInt(),
    );
  }
}

File: compose_recipe/app/src/main/java/com/example/recipeapp/data/mapper/IngredientMapper.kt
package com.example.recipeapp.data.mapper

import com.example.recipeapp.data.dto.IngredientDto
import com.example.recipeapp.data.model.Ingredient

fun IngredientDto.toModel(): Ingredient {
    return Ingredient(
        id = this.id!!,
        name = this.name!!,
        image = this.image!!
    )
}

fun Ingredient.toDto(): IngredientDto {
    return IngredientDto(
        id = this.id,
        name = this.name,
        image = this.image
    )
}